### DISCUSSION TITLE
Some Questions about EffectTask with lib tag 0.44.1?

### DISCUSSION THREAD

**Discussion by ACFancy** [pointfreeco/swift-composable-architecture#1709] [Q&A] [answered]
- Some usable functions like `EffectPublisher.timer(){} ` was marked as deprecated?
- How to migrate `Effect.merge Effect.concatenate` to `EffectTask.task(){} EffectTask.run(){} EffectTask.fireAndForget(){}`static functions? Whether is Needed to migrate?
- Is it necessary using Swift Concurrency,  `EffectTask.task(){} EffectTask.run(){} EffectTask.fireAndForget(){}` must be called in the end to return `EffectTask<Action, Never>` or `EffectTask<Action, Failure>` ?

**Comment by tgrapperon** [comment, marked as answer]
Hey @ACFancy!
If you look at #1477, you will see that `Combine` APIs will be slowly phased out from TCA to the profit of Swift concurrency. As a result, you're invited to use async/await variants when possible, and `Combine`-flavored APIs have been "softly" deprecated. They don't produce any warnings, but they're not proposed by Xcode code completion anymore.

You're not forced to migrate `Effect.merge` and `Effect.concatenate`, as these operations are already supported by concurrent API's. The corresponding `EffectTask`s will be executed concurrently when merging, or sequentially when concatenating.

`EffectTask` has only one "free" generic: `Action`. The failure is "hardcoded" to always be `Never`. This is a temporary name. In the future, and as described in #1477, it will be renamed `Effect`, and will only have generic and no failure at all. If you migrated your codebase to `EffectTask`, you will only have to rename them as `Effect`. `EffectPublisher` (which has two generics) will stay `EffectPublisher`, but will not be the basis of `Effect` anymore.

The fundamental `EffectTask` is `EffectTask.run`, which is an `EffectTask` that can send an arbitrary number of `action`s back into the system using the `send` argument. `EffectTask.task` is a specialization that sends (returns) exactly one action, and `EffectTask.fireAndForget`, a specialization that sends (returns) no action. When writing a reducer using Swift concurrency, you can return any of these `EffectTask` or `.none`.

Please let me know if these explanations are not clear or not answering to your questions.
